client/rest-high-level/src/main/java/org/elasticsearch/client/WatcherRequestConverters.java

```diff
@@ -108,6 +112,30 @@ static Request deleteWatch(DeleteWatchRequest deleteWatchRequest) {
         return request;
     }
 
+    static Request executeWatch(ExecuteWatchRequest executeWatchRequest) throws IOException {
+        RequestConverters.EndpointBuilder builder = new RequestConverters.EndpointBuilder()
+            .addPathPart("_xpack", "watcher", "watch");
+        if (executeWatchRequest.getId() != null) {
+            builder.addPathPart(executeWatchRequest.getId());
```

Comment:
addPathPart wont "build" anything thats null, so u can just call it instead of checking for null.

Reply:
++